src/main/java/uk/gov/mca/beacons/service/configuration/OpenApiConfig.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.mca.beacons.service.configuration;
+
+import io.swagger.v3.oas.models.ExternalDocumentation;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.info.Info;
+import io.swagger.v3.oas.models.info.License;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class OpenApiConfig {
+
+  private static final String GITHUB_LICENSE_SUFFIX = "/blob/main/LICENSE";
+
+  @Bean
```

Comment:
@zackads @Ninamma FYI. As discussed yesterday @zackads , `@Bean` annotation is a way of defining a __Spring managed__ bean.  What this essentially means, is that when the Spring application starts up it scans the directories for particular beans (java classes), that can be used at runtime.  When you annotate a class with:

- `@Controller`/`@RestController`
- `@Service`
- `@Component`
- `@Repository`

They are all annotations, which essentially adds metadata to each class, that Spring then uses to do things like [Dependency Injection](https://en.wikipedia.org/wiki/Dependency_injection), configuring/exposing REST endpoints, configuring data sources, security - a lot of things haha.

What we're doing here is defining a bean of type `OpenAPI` that the openapi dependency uses and injects into the Swagger UI/OpenAPI docs that Spring exposes for us.  A lot of the time when you are developing using Spring you usually don't worry about things like: `How does A get injected to B`, or `How does A know about this configuration defined over in this file`, the framework (Spring), manages all of that for you.

Excited to get started on some backend development next sprint and hopefully some of this starts to click into place 😄